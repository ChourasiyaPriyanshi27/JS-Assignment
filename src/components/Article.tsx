import React from "react";
import Marquee from "react-fast-marquee";
import axios from "axios";
import { Link } from "react-router-dom";
import { Card, Row } from "react-bootstrap";

class Article extends React.Component {
  constructor(props: any) {
    super(props);
    this.state = {
      headList: [],
      isLoading: true,
      searchText: "",
      text: [],
    };
  }

  componentDidMount() {
    axios
      .get(
        "https://newsapi.org/v2/top-headlines?country=us&apiKey=69f8347f69d8489dbad11bbfdc706156",
        {
          headers: {
            Accept: "application/json",
            Authorization: "Bearer 69f8347f69d8489dbad11bbfdc706156",
          },
        }
      )
      .then((res) => {
        this.setState({ headList: res.data.articles });
      });
  }

  render() {
    const { headList }: any = this.state;
    return (
      <div>
        <ul>
          <Marquee>
            {headList.map((auth: any, index: any) => (
              <Row className="justify-content-md-center" key={index}>
                <div>
                  <br />
                  <Card className="text-center" style={{ width: "18rem" }}>
                    <Link to={`/article/${index}`}>
                      <Card.Header>Top News: {auth.source.name}</Card.Header>
                    </Link>
                    <Card.Img
                      variant="top"
                      src={auth.urlToImage}
                      style={{ width: "18rem", height: "10rem" }}
                    />
                  </Card>
                </div>
              </Row>
            ))}
          </Marquee>
        </ul>
      </div>
    );
  }
}

export default Article;
