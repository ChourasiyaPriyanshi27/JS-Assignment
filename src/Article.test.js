import { Article } from "./Article";
import {shallow,mount} from 'enzyme'
import Table from "./Table";

describe("Article Page" , () => {
    it("click count" , () => {
        let wrapper = shallow(<Article />)
        expect(wrapper.find('.clicks-1').length).toEqual(1);
        // wrapper.find('button').simulate('click');
        // expect(wrapper.find('.clicks-2')).toEqual(2);
    })

    it("dive" , () => {
        let wrapper = shallow(<Article />)
        expect(wrapper.find(Table).dive().find('.custom-border').length).toBe(1)
    })
    
    it("sanpshot" , () => {
        let wrapper = shallow(<Table />)
        expect(wrapper).toMatchSnapshot()
    })
})

describe("Mount" , () => {
      
    it("Mount" , () => {
        let wrapper = mount(<Article />)
        console.log(wrapper.debug())
    })
})