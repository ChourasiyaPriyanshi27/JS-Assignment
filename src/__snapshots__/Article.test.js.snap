// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Page sanpshot 1`] = `
<table
  className="custom-border"
>
  <thead>
    <tr>
      <th>
        Names
      </th>
      <th>
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        NA
      </td>
    </tr>
  </tbody>
</table>
`;
